File: x10.runtime/src-java/x10/core/Ref.java
/*
 *  This file is part of the X10 project (http://x10-lang.org).
 *
 *  This file is licensed to You under the Eclipse Public License (EPL);
 *  You may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *      http://www.opensource.org/licenses/eclipse-1.0.php
 *
 *  (C) Copyright IBM Corporation 2006-2014.
 */

package x10.core;

import java.io.Serializable;

/**
 * Abstract implementation-level superclass of all X10 classes.
 */
@SuppressWarnings("serial")
public abstract class Ref implements Any, Serializable {
    
    // N.B. this is called implicitly by all subclasses of Ref
    public Ref() {}

    // constructor just for allocation
    public Ref(java.lang.System[] $dummy) {}

    // constructor for non-virtual call
    public final Ref x10$lang$Object$$init$S() {return this;}

    @Override
    public String toString() {
        return x10.lang.System.identityToString(this);
    }
 
// not used (same as Java)
//    @Override
//    public int hashCode() {
//        return x10.lang.System.identityHashCode$O(this);
//    }
//
//    @Override
//    public boolean equals(Object other) {
//        return x10.lang.System.identityEquals$O(this, other);
//    }
}

File: java-interop/x10.runtime/src-cpp/x10aux/ref.cc
/*
 *  This file is part of the X10 project (http://x10-lang.org).
 *
 *  This file is licensed to You under the Eclipse Public License (EPL);
 *  You may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *      http://www.opensource.org/licenses/eclipse-1.0.php
 *
 *  (C) Copyright IBM Corporation 2006-2010.
 */

#include <x10aux/config.h>

#include <x10aux/ref.h>
#include <x10aux/network.h>
#include <x10aux/throw.h>
#include <x10/lang/NullPointerException.h>
#include <x10/lang/BadPlaceException.h>
#include <x10aux/reference_logger.h>


using namespace x10aux;
using namespace x10::lang;

// do not call this if NO_EXCEPTIONS is defined
// defined here because it depends on NullPointerException and we don't want a header cycle
void x10aux::throwNPE() { throwException<NullPointerException>(); }

// vim:tabstop=4:shiftwidth=4:expandtab
